librad/src/profile.rs

```diff
@@ -91,83 +101,31 @@ impl Profile {
     /// ID is generated and written to `<root>/active_profile`.
     ///
     /// [`Paths::from_root`] for more information.
-    pub fn from_root(root: &Path, profile_id: Option<String>) -> Result<Self, Error> {
-        let id = if let Some(profile_id) = profile_id {
-            profile_id
-        } else {
-            load_profile_id(root)?
+    pub fn from_home(home: &RadHome, profile_id: Option<ProfileId>) -> Result<Self, Error> {
+        let id = match profile_id {
+            Some(id) => id,
+            None => ProfileId::load(home)?,
         };
 
-        let paths = Paths::from_root(root.join(&id))?;
+        let paths = match home {
+            RadHome::XDG => Paths::new(&id.0)?,
+            RadHome::Root(root) => Paths::from_root(root.join(&id.0))?,
+        };
 
         Ok(Self { id, paths })
     }
 
+    pub fn from_root(root: &Path, profile_id: Option<ProfileId>) -> Result<Self, Error> {
```

Comment:
Is this method strictly necessary? The `from_home` method seems to be sufficient for both variants of the `RadHome` enum, which both are constructible by callers.

Reply:
It's there to be backward-compatible and a lot of the testing of this uses `from_root` so it seems handy to have.